AWS-CodePipeline/two-environment-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,814 @@
+  {%- if cookiecutter.codestar_full_repository_id %}
+  FullRepositoryId:
+    Type: String
+    Default: "{{cookiecutter.codestar_full_repository_id}}"
+  {%- endif %}
+  {%- if cookiecutter.codecommit_repository_name %}
+  CodeCommitRepositoryName:
+    Type: String
+    Default: "{{cookiecutter.codecommit_repository_name}}"
+  {%- endif %}
+  MainGitBranch:
+    Type: String
+    Default: "{{cookiecutter.main_git_branch}}"
+  SamTemplate:
+    Type: String
+    Default: "{{cookiecutter.sam_template}}"
+  TestingRegion:
+    Type: String
+    Default: "{{cookiecutter.testing_region}}"
+  ProdRegion:
+    Type: String
+    Default: "{{cookiecutter.prod_region}}"
+  TestingStackName:
+    Type: String
+    Default: "{{cookiecutter.testing_stack_name}}"
+  ProdStackName:
+    Type: String
+    Default: "{{cookiecutter.prod_stack_name}}"
+  TestingPipelineExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.testing_pipeline_execution_role}}"
+  ProdPipelineExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.prod_pipeline_execution_role}}"
+  TestingCloudFormationExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.testing_cloudformation_execution_role}}"
+  ProdCloudFormationExecutionExeRole:
+    Type: String
+    Default: "{{cookiecutter.prod_cloudformation_execution_role}}"
+  TestingArtifactBucket:
+    Type: String
+    Default: "{{cookiecutter.testing_artifacts_bucket}}"
+  ProdArtifactBucket:
+    Type: String
+    Default: "{{cookiecutter.prod_artifacts_bucket}}"
+  TestingImageRepository:
+    Type: String
+    {%- if cookiecutter.testing_image_repository %}
+    Default: "{{cookiecutter.testing_image_repository}}"
+    {%- else %}
+    # If there are functions with "Image" PackageType in your template,
+    # Update the line below with image repository URL and add "--image-repository ${TESTING_IMAGE_REPOSITORY}" to
+    # prod "sam package" and "sam deploy" commands.
+    Default: ""
+    {%- endif %}
+  ProdImageRepository:
+    Type: String
+    {%- if cookiecutter.prod_image_repository %}
+    Default: "{{cookiecutter.prod_image_repository}}"
+    {%- else %}
+    # If there are functions with "Image" PackageType in your template,
+    # Update the line below with image repository URL and add "--image-repository ${PROD_IMAGE_REPOSITORY}" to
+    # prod "sam package" and "sam deploy" commands.
+    Default: ""
+    {%- endif %}
+  # CodeStarConnectionArn and FeatureGitBranch are required for pipelines for feature branches
+  CodeStarConnectionArn:
+    Type: String
+    Default: ""
+  FeatureGitBranch:
+    Type: String
+    Default: ""
+
+Conditions:
+  IsMainBranchPipeline: !Equals [!Ref FeatureGitBranch, ""]
+  IsFeatureBranchPipeline: !Not [Condition: IsMainBranchPipeline]
+  {%- if cookiecutter.codestar_full_repository_id %}
+  CreateConnection: !Equals [!Ref CodeStarConnectionArn, ""]
+  {%- endif %}
+
+Resources:
+  {%- if cookiecutter.codestar_full_repository_id %}
```

Comment:
nit: is it better to define two Jinja2 variables, something like:
1. {% set is_codestar_source = cookiecutter.codestar_full_repository_id != "" -%}
2. {% set is_codecommit_source = cookiecutter.codecommit_repository_name != "" -%}

Reply:
good idea, updated!